File: 7CM4 - DESARROLLO DE APLICACIONES MOVILES/Parcial 1/practicas/Practica 1 - Suma de Numeros/app/src/main/java/com/example/suma/MainActivity.java
package com.example.suma;

import android.os.Bundle;
import android.app.Activity;
import android.widget.*;

public class MainActivity extends Activity {
    int x, y, z;
    TextView jtv;

    protected void onCreate(Bundle b) {
        super.onCreate(b);
        setContentView(R.layout.activity_main);
        x = 2;
        y = 2;
        z = x + y;
        jtv = (TextView) findViewById(R.id.xtv);
        jtv.setText(x + "+" + y + "=" + z);
    }
}

File: 7CM4 - DESARROLLO DE APLICACIONES MOVILES/Parcial 1/practicas/Practica 7/app/src/main/java/com/example/intentoskotlin/MainActivity.kt
package com.example.intentoskotlin

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val coefA = findViewById<EditText>(R.id.coefA)
        val coefB = findViewById<EditText>(R.id.coefB)
        val coefC = findViewById<EditText>(R.id.coefC)
        val solveButton = findViewById<Button>(R.id.solveButton)

        solveButton.setOnClickListener {
            val a = coefA.text.toString().toDoubleOrNull()
            val b = coefB.text.toString().toDoubleOrNull()
            val c = coefC.text.toString().toDoubleOrNull()

            if (a != null && b != null && c != null) {
                val intent = Intent(this, ResultActivity::class.java).apply {
                    putExtra("coefA", a)
                    putExtra("coefB", b)
                    putExtra("coefC", c)
                }
                startActivity(intent)
            } else {
                coefA.error = "Por favor, ingresa un número válido"
                coefB.error = "Por favor, ingresa un número válido"
                coefC.error = "Por favor, ingresa un número válido"
            }
        }
    }
}
